frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_user_view.dart

```diff
@@ -62,27 +62,47 @@ class SettingsUserView extends StatelessWidget {
 }
 
 @visibleForTesting
-class UserNameInput extends StatelessWidget {
+class UserNameInput extends StatefulWidget {
   final String name;
+
   const UserNameInput(
     this.name, {
     Key? key,
   }) : super(key: key);
 
+  @override
+  UserNameInputState createState() => UserNameInputState();
+}
+
+class UserNameInputState extends State<UserNameInput> {
+  late TextEditingController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = TextEditingController(text: widget.name);
+  }
+
   @override
   Widget build(BuildContext context) {
     return TextField(
-      controller: TextEditingController()..text = name,
+      controller: _controller,
       decoration: InputDecoration(
         labelText: LocaleKeys.settings_user_name.tr(),
       ),
-      onSubmitted: (val) {
+      onChanged: (val) {
```

Comment:
Yes, this works, but it's expensive since we call the backend on every keystroke. Could you find a way to make that call only once? 

> https://stackoverflow.com/questions/51219330/flutter-run-function-after-showdialog-is-dismissed

Reply:
Or using `debounce`?

Reply:
Hey @a-wallen, I noticed that there were several commits on a single file with no significant changes. Additionally, I had to revert the `pre-commit.disabled` and `go-gitlint_1.1.0_linux_x86_64.tar.gz.1` files. Adding debounce would also result in more unnecessary commits. To keep the branch clean, I restored it to its initial state and continued from where I stopped.